Desktop_Interface/functiongencontrol.h

```diff
@@ -13,25 +13,26 @@
 
 //functionGenControl is a centralised object to control all of the high-level function gen commands for both channels.
 
+enum class ChannelID
```

Comment:
I know I said it would be best to move these out of the `functionGenControl`, but looking at it again I think it actually makes sense to keep them  namespaced.  "Channel", in the context of the Labrador, could refer to a signal generator, oscilloscope or logic analyzer channel, and all of these have different data.

Reply:
How about adding a `functionGen` namespace and renaming the `functionGenControl` class to `Controller`?

It is not a good fit with how the rest of the codebase is structured but I think it's worth a shot.

Then, we could have `functionGen::Controller`, `functionGen::ChannelID`, and `functionGen::ChannelData` which, inside the namespace would just be `Controller`, `ChannelID`, and `ChannelData`.

This looks like a good pattern to me because there are other places in the codebase that could use a `ChannelID` that shouldn't be implicitly convertible to `functionGen::ChannelID` but are otherwise identical.

There is also the thing that you cant do `using class::member_type;` like you can do with explicit namespaces, which then leads to awkwardness like `using member_type = class::member_type;`